### ISSUE TITLE
Implement vault and secrets

### ISSUE THREAD

**Issue by chouseknecht** [ansible/ansible-container#618]
@ehelms, @ryansb 

This is to track the progress, and facilitate further discussion, as we implement vault and secrets. 

The [proposal for implementing vault and secrets lives here](https://github.com/ansible/ansible-container/blob/develop/proposals/secrets.md).

@ehelms has agreed to take on the Ansible Container changes. @ryansb is working on upgrading the `docker_service` module, and @chouseknecht is working on creating a `docker_secret` module.

**Comment by chouseknecht** [comment]
The `docker_secret` module is finally ready. The delay was getting integration tests working for Docker modules in general. Thanks to the handy work of @mattclay we now have tests for the new module, and we'll be able to add tests for the others.

PR [#26469](https://github.com/ansible/ansible/pull/26469) containing the new module was merged.

**Comment by chouseknecht** [comment]
The code for generating Docker compose with secrets is complete. At least, it generates a playbook that works using the `docker_secret` and `docker_service` modules, and `ansible-container run` will successfully launch a test application. However, I'm not yet able to get an external secret to actually manifest in the running container. It should appear under `/run/secrets`.

**Comment by chouseknecht** [comment]
Figured out why compose is not mounting secrets to a running container. Turns out `docker-compose up` does not support external secrets. I took the generated compose from the deployment playbook, attempted a manual run, and got the following:

```
$ docker-compose up

WARNING: Service "web" uses secret "web_secret_password" which is external. External secrets are not available to containers created by docker-compose.
```

Apparently this only works with `docker stack deploy`.

**Comment by chouseknecht** [comment]
In the spirit of [docker compose #4368](https://github.com/docker/compose/pull/4368), which *fakes* secrets by bind mounting files, I added code that *fakes* external secrets by using a named volume. It's not glamorous, but it works. 

@ehelms, can you have a look, and give it a thumbs up or down. Thanks!

**Comment by chouseknecht** [comment]
Merged new `docker_volume` module to ansible. Using it when the `destroy` tag is invoked on the playbook generated by the `docker` engine.

**Comment by chouseknecht** [comment]
Closed by #665
